src/langsmith/diagnostics-self-hosted.mdx

```diff
@@ -0,0 +1,184 @@
+
+### Key deployed components
+
+Your deployment includes the following core components:
+
+- **`langsmith-frontend`**: The LangSmith frontend UI where you create Agent Server deployments. This app makes API calls to `langsmith-host-backend`. Part of the [control plane](/langsmith/control-plane).
+- **`langsmith-host-backend`**: The LangSmith Deployment [control plane](/langsmith/control-plane) that receives requests from `langsmith-frontend` and persists deployment requests to the control plane Postgres database.
+- **`langsmith-listener`**: Part of the LangSmith Deployment [data plane](/langsmith/data-plane). Polls `langsmith-host-backend` via HTTP API for deployments to create, update, or delete. Enqueues tasks for worker processes to handle.
+- **`langsmith-redis`**: The [Redis](/langsmith/data-plane#redis) instance serving as the task queue for `langsmith-listener`. The listener enqueues tasks here and workers pull tasks from this queue.
+- **`langsmith-operator`**: The `lgps` Kubernetes operator that reconciles underlying Kubernetes resources for `lgps` resources. Part of the data plane infrastructure.
+
+<Note>
+This list is not exhaustive. Additional components may be present in your deployment depending on your configuration. For a complete overview, refer to [LangSmith Deployment components](/langsmith/components).
+</Note>
+
+### Review application logs
+
+Tail the logs of each pod to understand baseline behavior:
+
+```bash
+kubectl logs -f <pod_name>
+```
+
+#### What to look for in logs
+
+First, enable `DEBUG` level logs (see Step 2 below). Then look for these log lines:
+
+- **`langsmith-listener`**: `Reconciling projects...` (appears every 10 seconds)
+- **`langsmith-operator`**: `Starting reconciliation` (appears periodically)
+
+In a healthy deployment, you should not see any errors. All logs should appear normal and routine.
+
+### Next steps
+
+Once you understand the baseline state, create a deployment from the LangSmith Frontend. This process will create an `lgps` Kubernetes resource that you can monitor.
+
+## Step 2. Enable debug logging
+
+When troubleshooting issues, the first step is typically to enable debug-level logging to gather more detailed information about what's happening in your system.
+
+### For control plane or data plane deployments
+
+If you are experiencing issues with a control plane deployment (for example, `langsmith-host-backend`) or a data plane deployment (for example, `langsmith-listener`), reinstall the Helm chart with the `LOG_LEVEL=DEBUG` environment variable. Add the following to your `values.yaml` file:
+
+```yaml
+extraEnv:
+  - name: LOG_LEVEL
+    value: DEBUG
+```
+
+### For Agent Server deployments
+
+If the issue is with an individual Agent Server deployment:
+
+1. Navigate to the **Deployments** tab in the [LangSmith UI](https://smith.langchain.com).
+1. On a deployment's view, select **+ New Revision**.
+1. Set the **Environment Variables** to `LOG_LEVEL` and the value to `DEBUG`.
```

Comment:
Suggestion: `Set the Environment Variables to LOG_LEVEL and the value to DEBUG.` --> `Add a new environment variable LOG_LEVEL and set it to DEBUG`.